foreman/data_refinery_foreman/foreman/main.py

```diff
@@ -50,14 +51,19 @@
 # This can be overritten by the env var "MAX_TOTAL_JOBS"
 DEFAULT_MAX_JOBS = 20000
 
-DOWNLOADER_JOBS_PER_NODE = 50
-PAGE_SIZE=2000
+PAGE_SIZE = 2000
 
-# This is the maximum number of non-dead nomad jobs that can be in the
-# queue. Set the default to one node's worth so we never are unable
-# to queue downloader jobs.
-MAX_TOTAL_DOWNLOADER_JOBS = DOWNLOADER_JOBS_PER_NODE
-TIME_OF_LAST_SIZE_CHECK = timezone.now() - datetime.timedelta(minutes=10)
+# The number of jobs running on each currently running volume
+VOLUME_WORK_DEPTH = dict()
+TIME_OF_LAST_WORK_DEPTH_CHECK = timezone.now() - datetime.timedelta(minutes=10)
+
+# The number of downloader jobs currently in the queue
+DOWNLOADER_JOBS_IN_QUEUE = 0
+TIME_OF_LAST_DOWNLOADER_JOB_CHECK = timezone.now() - datetime.timedelta(minutes=10)
+
+# The desired number of active + pending jobs on a volume. Downloader jobs
+# will be assigned to instances until this limit is reached.
+DESIRED_WORK_DEPTH = 300
```

Comment:
I think this number is too low. Right now with 2 8-weighted instances up we have 381 jobs running simultaneously. If we had a 20-weighted instance up it seems reasonable that it could probably run ~500 jobs itself. Therefore in order to have a bit of padding I'm thinking that we should set this to 1000. Even if we were at full capacity with 12 8-weighted instances we'd still only have a 12k job work queue depth, which is well under the 20k limit that the size of our nomad instances can handle.

Reply:
That makes sense. I didn't pick 300 for any particular reason, figuring that we would tune it later.